ygen/enumgen.go

```diff
@@ -363,6 +422,61 @@ func findEnumSet(entries map[string]*yang.Entry, compressPaths, noUnderscores, s
 	// where there are erroneous config/state differences).
 	sort.Strings(enumPaths)
 
+	s := newEnumGenState()
+
+	// This is the first of two passes over the input enum entries. The
+	// purpose of this pass is to resolve the names of all the enumerated
+	// values. During this pass, the identity names are fully resolved, and
+	// enumeration names are resolved pending deduplication.
+	for _, eP := range enumPaths {
+		e := validEnums[eP]
+		_, builtin := yang.TypeKindFromName[e.Type.Name]
+		switch {
+		case e.Type.Name == "union", len(e.Type.Type) > 0 && !builtin:
+			// Calculate any enumerated types that exist within a union, whether it
+			// is a directly defined union, or a non-builtin typedef.
+			if err := s.resolveEnumeratedUnionEntry(e, compressPaths, noUnderscores, skipEnumDedup); err != nil {
+				errs = append(errs, err)
+			}
+		case e.Type.Name == "identityref":
+			// This is an identityref - we do not want to generate code for an
+			// identityref but rather for the base identity. This means that we reduce
+			// duplication across different enum types. Re-map the "path" that is to
+			// be used to the new identityref name.
+			if e.Type.IdentityBase == nil {
+				errs = append(errs, fmt.Errorf("entry %s was an identity with a nil base", e.Name))
+				continue
+			}
+			if err := s.resolveIdentityRefBaseType(e, noUnderscores); err != nil {
+				errs = append(errs, err)
+			}
+		case e.Type.Name == "enumeration":
+			// We simply want to map this enumeration into a new name. Since we do
+			// de-duplication of re-used enumerated leaves at different points in
+			// the schema (e.g., if openconfig-bgp/container/enum-A can be instantiated
+			// in two places, then we do not want to have multiple enumerated types
+			// that represent this leaf), then we do not have errors if duplicates
+			// occur, we simply perform de-duplication at this stage.
```

Comment:
Is this comment now stale -- since we're explicitly not de-duplicating at this stage right?